File: basketgame-android/app/src/main/java/com/basket_game/Equipe.java
/**
 * @file Equipe.java
 * @brief Déclaration de la classe Equipe
 * @author Guillaumet Florent
 */

package com.basket_game;

import android.util.Log;

import java.io.Serializable;

/**
 * @class Equipe
 * @brief Définit une équipe
 */
public class Equipe implements Serializable
{
    /**
     * Constantes
     */
    private static final String TAG = "_Equipe"; //!< TAG pour les logs (cf. Logcat)

    /**
     * @brief Attributs
     */
    private String nomEquipe; //!< le nom de l'équipe
    public int     scorePanier;
    public int     scoreManche;

    /**
     * @brief Constructeur de la classe Equipe
     */
    public Equipe()
    {
        Log.d(TAG, "Equipe()");
        this.nomEquipe = "";
        this.scorePanier     = 0;
        this.scoreManche     = 0;
    }

    /**
     * @brief Constructeur de la classe Equipe
     */
    public Equipe(String nomEquipe)
    {
        Log.d(TAG, "Equipe(" + nomEquipe + ")");
        this.nomEquipe = nomEquipe;
    }

    /**
     *
     * @brief Accesseur Equipe
     * @fn getNomEquipe
     * @details Retourne le nom
     * @return String
     */
    public String getNomEquipe()
    {
        return nomEquipe;
    }

    /**
     *
     * @brief Mutateur Equipe
     * @fn setNomEquipe
     * @details Modifie le nom de l'Equipe
     * @param nomEquipe
     */
    public void setNomEquipe(String nomEquipe)
    {
        this.nomEquipe = nomEquipe;
    }

    /**
     * @brief Accesseur Equipe
     * @fn getScorePanier
     * @details Retourne le nombre de panier inscrits
     * @return int
     */
    public int getScorePanier()
    {
        return scorePanier;
    }

    /**
     * @brief Méthode appelée pour réinitialiser le nombre de panier des équipes
     */
    public void reinitialiserScorePanier()
    {
        this.scorePanier = 0;
    }

    /**
     * @brief Méthode appelée pour incrémenter de 1 le scorePanier
     */
    public void incrementerScorePanier()
    {
        this.scorePanier++;
    }

    /**
     * @brief Accesseur Equipe
     * @fn getScoreManche
     * @details Retourne le nombre de manche gagnée par l'équipe
     * @return int
     */
    public int getScoreManche()
    {
        return scoreManche;
    }

    /**
     * @brief Méthode appelée pour réinitialiser le nombre de manche gagnée
     */
    public void reinitialiserScoreManche()
    {
        this.scoreManche = 0;
    }

    /**
     * @brief Méthode appelée pour incrémenter de 1 le nombre de manche gagnée
     */
    public void incrementerScoreManche()
    {
        this.scoreManche++;
    }
}

File: BasketGameUI/equipe.cpp
/**
 * @file equipe.cpp
 *
 * @brief Définition de la classe Equipe
 * @author Nathanael CHANSARD
 * @version 1.0
 */

#include "equipe.h"
#include "puissance4.h"
#include <QDebug>

/**
 * @brief Constructeur de la classe Equipe
 *
 * @fn Equipe::Equipe
 *
 */
Equipe::Equipe(const QString& nom, QObject* parent) :
    QObject(parent), nom(nom), scorePanier(0) , scoreManche(0)
{
    qDebug() << Q_FUNC_INFO << "nom" << nom;
}

/**
 * @brief Destructeur de la classe Equipe
 *
 * @fn Equipe::~Equipe
 *
 */
Equipe::~Equipe()
{
    qDebug() << Q_FUNC_INFO;
}


/**
 * @fn Equipe::getNom()
 * @brief Retourne le nom de l'équipe
 */
QString Equipe::getNom() const
{
    return nom;
}

/**
 * @fn Equipe::setNom()
 * @brief Modifie le nom de l'équipe
 */
void Equipe::setNom(const QString& nom)
{
    this->nom = nom;
    qDebug() << Q_FUNC_INFO << "nom" << nom;
}

/**
 * @fn Equipe::getScore()
 * @brief Retourne le score de l'équipe
 */
int Equipe::getScorePanier() const
{
    return scorePanier;
}

/**
 * @fn Equipe::setScore()
 * @brief  Modifie le score de l'équipe
 */
void Equipe::setScorePanier(int scorePanier)
{
    this->scorePanier = scorePanier;
    qDebug() << Q_FUNC_INFO << "score" << scorePanier;
}
/**
 * @fn Equipe::getScoreManche()
 * @brief Retourne le score de l'équipe
 */
int Equipe::getScoreManche() const
{
    return scoreManche;
}

void Equipe::setScoreManche(int scoreManche)
{
    this->scoreManche = scoreManche;
    qDebug() << Q_FUNC_INFO << "scoreManche" << scoreManche;

}

/**
 * @fn Equipe::incrementerScore()
 * @brief méthode pour incrementer le score de chaque équipe
 */
void Equipe::incrementerScorePanier()
{
    scorePanier++;
    qDebug() << Q_FUNC_INFO << "score" << scorePanier;
}

void Equipe::incrementerScoreManche()
{
    scoreManche++;
}
